File: src/main/java/k2b6s9j/BoatCraft/registry/EntityRegistry.java
package k2b6s9j.BoatCraft.registry;

import k2b6s9j.BoatCraft.BoatCraft;
import k2b6s9j.BoatCraft.entity.boat.wood.birch.*;
import k2b6s9j.BoatCraft.entity.boat.wood.jungle.*;
import k2b6s9j.BoatCraft.entity.boat.wood.oak.*;
import k2b6s9j.BoatCraft.entity.boat.wood.spruce.*;

public class EntityRegistry {

    static BoatCraft mod;

    public static void RegisterEntities() {
        int id = 1;
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatOak.class, "Oak Wood Boat", id, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatOakChest.class, "Oak Wood Chest Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatOakFurnace.class, "Oak Wood Furnace Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatOakHopper.class, "Oak Wood Hopper Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatOakTNT.class, "Oak Wood TNT Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatSpruce.class, "Spruce Wood Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatSpruceChest.class, "Spruce Wood Chest Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatSpruceFurnace.class, "Spruce Wood Furnace Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatSpruceHopper.class, "Spruce Wood Hopper Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatSpruceTNT.class, "Spruce Wood TNT Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatBirch.class, "Birch Wood Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatBirchChest.class, "Birch Wood Chest Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatBirchFurnace.class, "Birch Wood Furnace Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatBirchHopper.class, "Birch Wood Hopper Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatBirchTNT.class, "Birch Wood TNT Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatJungle.class, "Jungle Wood Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatJungleChest.class, "Jungle Wood Chest Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatJungleFurnace.class, "Jungle Wood Furnace Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatJungleHopper.class, "Jungle Wood Hopper Boat", id++, mod, 80, 3, true);
        cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(EntityBoatJungleTNT.class, "Jungle Wood TNT Boat", id++, mod, 80, 3, true);
    }
}

File: src/main/scala/k2b6s9j/BoatCraft/registry/EntityRegistry.scala
package k2b6s9j.BoatCraft.registry

import k2b6s9j.BoatCraft.BoatCraft
import k2b6s9j.BoatCraft.boat.wood._

object EntityRegistry {

    def RegisterEntities() {
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[oak.Empty.Entity], "Oak Wood Boat", 1, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[oak.Chest.Entity], "Oak Wood Chest Boat", 2, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[oak.Furnace.Entity], "Oak Wood Furnace Boat", 3, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[oak.Hopper.Entity], "Oak Wood Hopper Boat", 4, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[oak.Hopper.Entity], "Oak Wood TNT Boat", 5, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[spruce.Empty.Entity], "Spruce Wood Boat", 6, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[spruce.Chest.Entity], "Spruce Wood Chest Boat", 7, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[spruce.Furnace.Entity], "Spruce Wood Furnace Boat", 8, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[spruce.Hopper.Entity], "Spruce Wood Hopper Boat", 9, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[spruce.TNT.Entity], "Spruce Wood TNT Boat", 10, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[birch.Empty.Entity], "Birch Wood Boat", 11, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[birch.Chest.Entity], "Birch Wood Chest Boat", 12, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[birch.Furnace.Entity], "Birch Wood Furnace Boat", 13, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[birch.Hopper.Entity], "Birch Wood Hopper Boat", 14, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[birch.TNT.Entity], "Birch Wood TNT Boat", 15, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[jungle.Empty.Entity], "Jungle Wood Boat", 16, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[jungle.Chest.Entity], "Jungle Wood Chest Boat", 17, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[jungle.Furnace.Entity], "Jungle Wood Furnace Boat", 18, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[jungle.Hopper.Entity], "Jungle Wood Hopper Boat", 19, BoatCraft, 80, 3, true)
      cpw.mods.fml.common.registry.EntityRegistry.registerModEntity(classOf[jungle.TNT.Entity], "Jungle Wood TNT Boat", 20, BoatCraft, 80, 3, true)
    }
}
